Integrations/Integration-Salesforce.yml

```diff
@@ -0,0 +1,174 @@
+configuration:
+- display: Credentials
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Consumer Key
+  name: client_id
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Consumer Secret
+  name: client_secret
+  defaultvalue: ""
+  type: 4
+  required: true
+script:
+  script: |-
+    var authUrl = 'https://login.salesforce.com/services/oauth2/token';
+    var uriPrefix = '/services/data/v20.0/';
+
+    function getNewToken() {
+        var request = {
+                grant_type: 'password',
+                client_id: params.client_id,
+                client_secret: params.client_secret,
+                username: params.credentials.identifier,
+                password: params.credentials.password
+            };
+
+        var body = encodeToURLQuery(request).substr(1);
+        var response = http(
+                    authUrl,
+                    {
+                        Method: 'POST',
+                        Headers: {'Content-Type': ['application/x-www-form-urlencoded']},
+                        Body: body,
+                    },
+                    false
+                );
+            if (response.StatusCode < 200 || response.StatusCode >= 300) {
+                throw 'Failed to get new token, request status code: ' + response.StatusCode + ' and Body: ' + response.Body + '.';
+            }
+        return JSON.parse(response.Body);
+    }
+
+    function updateSessionContext() {
+        var newResponse = getNewToken();
+        setIntegrationContext(newResponse);
+        return newResponse;
+    }
+
+
+    function sendRequest(method, url, body, token) {
+        var headers = {};
+        if (token) {
+            headers['Authorization'] = ['Bearer ' + token];
+        }
+        if(method == 'POST' || method == 'PATCH') {
+            headers['Content-Type'] = ['application/json'];
+        }
+            return http(
+                    url,
+                    {
+                        Method: method,
+                        Headers: headers,
+                        Body: body,
+                    },
+                    false
+                );
+        }
+
+    function sendRequestInSession(method, uri, body) {
+        var contextData = getIntegrationContext();
+        if (!contextData || !contextData.access_token) {
```

Comment:
View box for better handling in case of two updates in the same time.

I prefer KISS - keep it simple stupid and always authenticate